src/modalities/trainer.py

```diff
@@ -66,6 +66,8 @@ def train(
         cummulated_loss = self._reset_loss()
         thoughput_aggregator = Aggregator[ThroughputAggregationKeys]()
 
+        device = torch.device(self.local_rank) if torch.cuda.is_available() else "cpu"
```

Comment:
This line also appears in `src/modalities/evaluator.py` (line 49), we might want to use a function to avoid duplicate code.